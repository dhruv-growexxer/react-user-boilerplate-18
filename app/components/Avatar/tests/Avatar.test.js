import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { createMemoryHistory } from 'history';
import { HistoryRouter as Router } from 'redux-first-history/rr6';
import Avatar, { getNewMenu, returnItems } from '../index';
import configureStore from '../../../configureStore';

describe('<Avatar />', () => {
  const stubProps = {
    menu: [
      {
        to: '/profile',
        tabName: 'Profile',
      },
      {
        to: '/change-password',
        tabName: 'Change Password',
      },
      {
        to: '/logout',
        tabName: 'Logout',
      },
    ],
  };
  const history = createMemoryHistory();
  const { store } = configureStore({}, history);

  it('should render a div', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <Router history={history}>
            <Avatar menu={stubProps} />
          </Router>
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should returnItems and getNewMenu', () => {
    const items = [
      {
        to: '/profile',
        tabName: 'Profile',
      },
      {
        to: '/change-password',
        tabName: 'Change Password',
      },
    ];
    const currReturnItems = returnItems(items);
    const currGetNewMenu = getNewMenu(items);
    expect(currReturnItems).toBeTruthy();
    expect(currGetNewMenu).toBeTruthy();
  });
});
