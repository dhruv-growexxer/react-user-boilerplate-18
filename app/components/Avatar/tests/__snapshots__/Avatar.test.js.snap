// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render a div 1`] = `null`;
